eZ/Publish/API/Repository/Tests/TrashServiceTest.php

```diff
@@ -684,6 +685,41 @@ public function testFindTrashItems()
         $this->assertEquals(4, $searchResult->totalCount);
     }
 
+    /**
+     * Test for the findTrashItems() method.
+     *
+     * @see \eZ\Publish\API\Repository\TrashService::findTrashItems()
+     * @depends eZ\Publish\API\Repository\Tests\TrashServiceTest::testTrash
+     */
+    public function testFindTrashItemsSortedByDateTrashed()
+    {
+        $this->createTrashItem();
+        sleep(1); // make sure trashed date changed
```

Comment:
Hmm, the tests are long enough, we shouldn't create artificial overheads. Please notice that due to `...orEqual` calls later on, this line is irrelevant (the test passes w/o that).

What you should test here is if two known objects are loaded in the expected order, so it's enough to test their e.g. IDs (but not hardcoded). Moreover new tests should not rely on pre-existing data - it causes a lot of maintenance issues.

We lack global system clock which could be mutated during tests, so for now what you could do is raw database update:

```php
    /**
     * @covers \eZ\Publish\API\Repository\TrashService::findTrashItems
     *
     * @depends eZ\Publish\API\Repository\Tests\TrashServiceTest::testTrash
     *
     * @throws \ErrorException
     * @throws \eZ\Publish\API\Repository\Exceptions\ForbiddenException
     * @throws \eZ\Publish\API\Repository\Exceptions\NotFoundException
     * @throws \eZ\Publish\API\Repository\Exceptions\UnauthorizedException
     */
    public function testFindTrashItemsSortedByDateTrashed(): void
    {
        $repository = $this->getRepository();
        $trashService = $repository->getTrashService();
        $locationService = $repository->getLocationService();

        $folder1 = $this->createFolder(['eng-GB' => 'Folder1'], 2);
        $folder2 = $this->createFolder(['eng-GB' => 'Folder2'], 2);

        $firstTrashedItem = $trashService->trash(
            $locationService->loadLocation($folder1->contentInfo->mainLocationId)
        );
        $this->updateTrashedDate($firstTrashedItem->id, \time() - 100);
        $latestTrashItem = $trashService->trash(
            $locationService->loadLocation($folder2->contentInfo->mainLocationId)
        );

        $query = new Query();
        $query->filter = new Criterion\LogicalAnd([]);

        // Load all trashed locations, sorted by trashed date ASC
        $query->sortClauses = [new SortClause\Trash\DateTrashed(Query::SORT_ASC)];
        $searchResult = $trashService->findTrashItems($query);
        self::assertEquals(2, $searchResult->totalCount);
        self::assertEquals($firstTrashedItem->remoteId, $searchResult->items[0]->remoteId);
        self::assertEquals($latestTrashItem->remoteId, $searchResult->items[1]->remoteId);

        // Load all trashed locations, sorted by trashed date DESC
        $query->sortClauses = [new SortClause\Trash\DateTrashed(Query::SORT_DESC)];
        $searchResult = $trashService->findTrashItems($query);
        self::assertEquals(2, $searchResult->totalCount);
        self::assertEquals($latestTrashItem->remoteId, $searchResult->items[0]->remoteId);
        self::assertEquals($firstTrashedItem->remoteId, $searchResult->items[1]->remoteId);
    }

    /**
     * @throws \ErrorException
     */
    private function updateTrashedDate(int $locationId, int $newTimestamp): void
    {
        $connection = $this->getRawDatabaseConnection();
        $query = $connection->createQueryBuilder();
        $query
            ->update('ezcontentobject_trash')
            ->set('trashed', ':trashed_timestamp')
            ->where('node_id = :location_id')
            ->setParameter('trashed_timestamp', $newTimestamp, ParameterType::INTEGER)
            ->setParameter('location_id', $locationId, ParameterType::INTEGER);
        $query->execute();
    }
```

It ain't pretty, but faster and more reliable.